core/blockchain.go

```diff
@@ -2394,9 +2394,25 @@ func (bc *BlockChain) maintainTxIndex() {
 	}
 	defer sub.Unsubscribe()
 
+	// If geth is running in read-only mode(not receiving new blocks),
```

Comment:
Instead of introducing a timer, we can always spin up a indexBlocks action before loop, as the initialization step.

In the scenarios you mentioned, Geth should index/uninindex the unfinished chain segments and then do nothing(because chain has no progress)

Reply:
thanks for the advice.

Reply:
@rjl493456442 please take another look, thanks